src/realm/object-store/sync/sync_session.cpp

```diff
@@ -888,11 +888,18 @@ static sync::Session::Config::ClientReset make_client_reset_config(RealmConfig s
                                                           did_recover);
         };
     }
-    if (session_config.sync_config->notify_before_client_reset) {
-        config.notify_before_client_reset = [config = session_config](VersionID version) {
-            config.sync_config->notify_before_client_reset(Realm::get_frozen_realm(config, version));
-        };
-    }
+    bool notify_user_before = bool(session_config.sync_config->notify_before_client_reset);
+    config.notify_before_client_reset = [notify_user_before, config = session_config]() -> VersionID {
+        auto frozen = Realm::get_shared_realm(config)->freeze();
+        REALM_ASSERT_EX(frozen, config.path);
+        REALM_ASSERT(frozen->is_frozen());
+        util::Optional<VersionID> version = frozen->current_transaction_version();
```

Comment:
`read_transaction_version()` is the version of this that doesn't return an optional.

Reply:
I ran into an [assertion](https://github.com/realm/realm-core/blob/master/src/realm/object-store/shared_realm.cpp#L647) trying to use that method because in the case where opening the Realm here doesn't have any schema changes to make with the version on disk, then there is no active transaction. Oddly only `current_transaction_version()` does correctly handle the version for frozen Realms.